GitHub.Collectors.Functions/GitHubFunctions.cs

```diff
@@ -706,6 +710,116 @@ public async Task ExecutePointCollectorAsync(string queueItem, ExecutionContext
+                ICache<ConditionalRequestTableEntity> requestsCache = new AzureTableCache<ConditionalRequestTableEntity>(telemetryClient, "requests");
+                await requestsCache.InitializeAsync().ConfigureAwait(false);
+                GitHubHttpClient httpClient = new GitHubHttpClient(this.httpClient, rateLimiter, requestsCache, telemetryClient);
+                IAuthentication auth = this.configManager.GetAuthentication(CollectorType.DiscoverOrganizations, httpClient, null, this.apiDomain);
+
+                // check that collector functions are using github app authentication.
+                if (!(auth is GitHubAppAuthentication))
+                {
+                    telemetryClient.LogWarning("Discover Organizations requires that authentication be of type GitHubAppAuthentication.");
+                    return;
+                }
+
+                // get all organizations where the GitHub app is installed.
+                GitHubAppAuthentication githubAuth = (GitHubAppAuthentication)this.configManager.GetAuthentication(CollectorType.DiscoverOrganizations, httpClient, null, this.apiDomain);
+                JArray installations = await githubAuth.GetAppInstallations().ConfigureAwait(false);
+
+                // build configuration.json from organizations in JArray.
+                JArray formattedInstallations = new JArray();
+                Dictionary<int, string> discoveredOrganizationMap = new Dictionary<int, string>();
+                foreach (JObject installation in installations)
+                {
+                    JObject formattedInstallation = new JObject();
+                    formattedInstallation.Add("OrganizationLogin", installation.SelectToken("account.login"));
+                    formattedInstallation.Add("OrganizationId", installation.SelectToken("account.id"));
+                    formattedInstallations.Add(formattedInstallation);
+
+                    int id = installation.SelectToken("account.id").Value<int>();
+                    string login = installation.SelectToken("account.login").Value<string>();
+
+                    discoveredOrganizationMap[id] = login;
+                }
+
+                formattedInstallations = new JArray(formattedInstallations.OrderBy(obj => (string)obj["OrganizationLogin"]));
+                string configurationString = formattedInstallations.ToString();
+
+                // output to blob.
+                CloudBlockBlob outputBlob = AzureHelpers.GetBlob("github-settings", "generated-organizations.json");
+                CloudBlobStream cloudBlobStream = await outputBlob.OpenWriteAsync().ConfigureAwait(false);
+                StreamWriter writer = new StreamWriter(cloudBlobStream, Encoding.UTF8);
+                await writer.WriteLineAsync(configurationString).ConfigureAwait(false);
+                writer.Dispose();
```

Comment:
Can we encapsulate this in AzureHelpers as a helper method? Also, can you use "using" instead of explicit Dispose?